string resourceName
uint32 amountGranted
